api/swagger.yaml

```diff
@@ -0,0 +1,154 @@
+openapi: 3.0.0
+
+info:
+  title: "Speculos API"
+  version: "1.0.0"
+
+servers:
+  - url: "http://127.0.0.1:5000"
+
+paths:
+  /automation:
+    post:
+      summary: "Updates the automation rules"
+      externalDocs:
+        description: More information about the automation feature and the rules format can be found in the speculos documentation.
+        url: https://github.com/LedgerHQ/speculos/blob/master/docs/user/automation.md
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+            examples:
+              full-syntax:
+                summary: Regexp, conditions and actions
+                value: { "version": 1, "rules": [ { "text": "Application", "x": 35, "y": 3, "conditions": [ [ "seen", false ] ], "actions": [ [ "button", 2, true ], [ "button", 2, false ], [ "setbool", "seen", true ] ] }, { "regexp": "\\d+", "actions": [ [ "exit" ] ] }, { "actions": [ [ "setbool", "default_match", true ] ] } ] }
+              press-buttons:
+                summary: Press both button when the text "Approve" is displayed
+                value: {"version": 1, "rules": [ { "text": "Approve", "actions": [ [ "button", 1, true ], [ "button", 2, true ], [ "button", 1, false ], [ "button", 2, false ] ] } ] }
+      responses:
+        "200":
+          description: "successful operation"
+
+  /button/{button}:
+    post:
+      summary: "Press or release a button (Nano S and Nano X)"
+      parameters:
+      - name: "button"
+        description: "Button to press"
+        in: path
+        required: true
+        schema:
+          $ref: '#/components/schemas/Button'
+      - name: "action"
+        in: "query"
+        description: "Press or release the specified button. If unset, the button is pressed then released."
+        required: false
+        schema:
+          $ref: '#/components/schemas/Action'
+      - name: "delay"
+        in: "query"
+        description: "Delay in seconds between press and release, ignored if action is specified."
+        required: false
+        schema:
+          type: number
+          format: float
+          default: 0.1
+      responses:
+        "200":
+          description: "successful operation"
+        "400":
+          description: "invalid parameter"
+
+  /events:
+    get:
+      summary: "Get the events produced by the app"
+      parameters:
+      - name: "stream"
+        description: "Stream the events instead of returning a list"
+        in: query
+        required: false
+        schema:
+          type: boolean
+          default: true
```

Comment:
Using `stream=true` by default makes the Swagger web UI "freeze" when issuing a call to the `/events` endpoint. Using `false` by default in the web UI seems more user-friendly.

Reply:
done